api/swagger.yaml

```diff
@@ -0,0 +1,788 @@
+      - $ref: '#/components/schemas/RenameCommand'
+    MoveCommand:
+      allOf:
+      - $ref: '#/components/schemas/Command'
+      - required:
+        - id
+        - where
+        type: object
+        properties:
+          id:
+            type: integer
+            format: int64
+          where:
+            type: integer
+            format: int64
+    StopCommand:
+      allOf:
+      - $ref: '#/components/schemas/Command'
+      - required:
+        - id
+        type: object
+        properties:
+          id:
+            type: integer
+            format: int64
+    ConstructCommand:
+      allOf:
+      - $ref: '#/components/schemas/Command'
+      - required:
+        - id
+        - where
+        type: object
+        properties:
+          id:
+            type: integer
+            format: int64
+          where:
+            type: integer
+            format: int64
+    DecommissionCommand:
+      allOf:
+      - $ref: '#/components/schemas/Command'
+      - required:
+        - id
+        type: object
+        properties:
+          id:
+            type: integer
+            format: int64
+    RenameCommand:
+      allOf:
+      - $ref: '#/components/schemas/Command'
+      - required:
+        - id
+        - name
+        type: object
+        properties:
+          id:
+            type: integer
+            format: int64
+          name:
+            type: string
+    AttackCommand:
+      allOf:
+      - $ref: '#/components/schemas/Command'
+      - required:
+        - id
+        - target
+        type: object
+        properties:
+          id:
+            type: integer
+            format: int64
+          target:
+            type: integer
+            format: int64
+    TradeCommand:
+      allOf:
+      - $ref: '#/components/schemas/Command'
+      - required:
+        - amount
+        - id
+        - resource
+        - where
+        type: object
+        properties:
+          id:
+            type: integer
+            format: int64
+          resource:
+            type: integer
+            format: int64
+          amount:
+            type: integer
+            format: int64
+          where:
+            type: integer
+            format: int64
+    Commands:
+      type: object
+      properties:
+        commands:
+          type: array
+          items:
+            $ref: '#/components/schemas/Command'
+    Credentials:
+      required:
+      - name
+      - password
+      - player
+      type: object
+      properties:
+        name:
+          type: string
+        password:
+          type: string
+        player:
+          type: string
+    Tick:
+      type: object
+      properties:
+        tick:
+          type: integer
+          description: Game tick in milliseconds
+          format: int64
+          example: 10
+        wait:
+          type: boolean
+          default: true
+
+    Coordinates:
+      type: object
+      properties:
+        x:
```

Comment:
is it necessary to have coordinates as separate x, y? i think it would be easier to work with array with 2 elements.

Reply:
I don't see a way to define "array with exactly two elements" in OpenAPI so I would keep using `x` and `y` to make the API "more defined". E.g. array with one element would be valid in API but invalid in game logic. With separate `x` and `y` we are always sure that exactly two coordinates are present.

Reply:
https://stackoverflow.com/a/57036697/11322957

Reply:
I don't have a strong opinion, so why not ;)